src/components/button/button.stories.js

```diff
@@ -38,45 +30,128 @@ const getKnobs = () => {
-    knobs: {
-      escapeHTML: false
-    }
-  };
+asASibling.story = {
+  name: 'as a sibling'
+};
 
-  return [name, component, metadata];
-}
 
-function makeSiblingStory(name, themeSelector) {
-  const component = () => {
-    const props = getKnobs();
-    const { children } = props; // eslint-disable-line react/prop-types
-    return (
-      <div>
-        <Button { ...props }>{ children }</Button>
-        <Button { ...props }>{ children }</Button>
-      </div>
-    );
-  };
+export const allButtons = () => {
+  const types = ['primary', 'secondary', 'tertiary'];
+  const sizes = ['small', 'medium', 'large'];
+  return <>
+    {['before', 'after'].map(iconPosition => (
+      types.map(type => (
+        ['', ...OptionsHelper.icons].map(icon => (
+          <div>
+            {sizes.map(size => (
+              <>
+                <Button
+                  size={ size } iconPosition={ iconPosition }
+                  iconType={ icon }
+                  buttonType={ type }
+                >{size}
+                </Button>
 
-  const metadata = {
-    themeSelector
-  };
+                {size === 'large' && (
+                  <Button
+                    size={ size } iconPosition={ iconPosition }
+                    iconType={ icon }
+                    buttonType={ type }
+                    subtext='line two'
+                  >{size}
+                  </Button>
+                )}
+              </>
+            ))}
+
+            {sizes.map(size => (
+              <>
+                <Button
+                  iconPosition={ iconPosition }
+                  iconType={ icon }
+                  size={ size }
+                  destructive
+                  buttonType={ type }
+                >{size}
+                </Button>
 
-  return [name, component, metadata];
-}
+                {size === 'large' && (
+                  <Button
+                    iconPosition={ iconPosition }
+                    iconType={ icon }
+                    size={ size }
+                    destructive
+                    buttonType={ type }
+                    subtext='line two'
+                  >{size}
+                  </Button>
+                )}
+              </>
+            ))}
 
-storiesOf('Button', module)
-  .add(...makeStory('default', Info))
-  .add(...makeSiblingStory('as a sibling'));
+            {sizes.map(size => (
+              <>
+                <Button
+                  iconPosition={ iconPosition }
+                  iconType={ icon }
+                  size={ size }
+                  disabled
+                  buttonType={ type }
+                >{size}
+                </Button>
+
+                {size === 'large' && (
+                  <Button
+                    iconPosition={ iconPosition }
+                    iconType={ icon }
+                    size={ size }
+                    disabled
+                    buttonType={ type }
+                    subtext='line two'
```

Comment:
I think that you could extract `iconPosition, iconType, size and buttonType` to an object and just add these props as `{ ...objectName }`

Reply:
I've done it in a new commit, can you let me know what you think. I'm not sure if it helps the readability of the code. I did the other OptionHelper too, but again I think it makes the code harder to read.